### PR TITLE
UIDATIMP-724: Cover <Section> component with tests

### PR DESCRIPTION
**Pull request by OleksandrHladchenko1**
## Purpose

Cover `<Section>` component with tests using RTL+Jest.

## Approach

Covered `<Section>` component with tests using RTL+Jest.

## Refs

https://issues.folio.org/browse/UIDATIMP-724

## Screenshots

image

### CODE REVIEW

**Review by IvanKryzhanovskyi — COMMENTED** [review]
Add, please, correspond updates into changelog.md file

File: src/components/Section/Section.test.js [hunk 1]
```diff
@@ -0,0 +1,105 @@
+import React from 'react';
+
+import { renderWithIntl } from '@folio/stripes-data-transfer-components/test/jest/helpers';
+import '../../../test/jest/__mock__';
+import { fireEvent } from '@testing-library/dom';
```
Comment (mariia-aloshyna) [inline src/components/Section/Section.test.js:5]: Import `fireEvent` from `@testing-library/react`, please
And put this import under `import React from 'react'`

File: src/components/Section/Section.test.js [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    <span>child component</span>
+  </div>
+);
+const sectionWithLabel = {
+  label: 'test label',
+  optional: false,
+};
+const sectionWithLabelAndDataTestAttribute = {
+  label: 'test label',
+  rest: 'data-test-tester',
```
Comment (mariia-aloshyna) [inline src/components/Section/Section.test.js:21]:
```suggestion
  'data-test-tester': true,
```

File: src/components/Section/Section.test.js [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+};
+const sectionWithCheckboxAndLabel = {
+  label: 'test label',
+  optional: true,
+  isOpen: false,
+  children: [childElement],
+};
+
+const renderSectionContainer = ({
+  label, optional, children, isOpen, rest,
```
Comment (mariia-aloshyna) [inline src/components/Section/Section.test.js:39]: If there are more than one props, put each on a new line, please

File: src/components/Section/Section.test.js [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+  optional: false,
+};
+const sectionWithCheckboxAndLabel = {
+  label: 'test label',
+  optional: true,
+  isOpen: false,
+  children: [childElement],
+};
+
+const renderSectionContainer = ({
```
Comment (mariia-aloshyna) [inline src/components/Section/Section.test.js:38]: Name it like in other tests for consistency, please
```suggestion
const renderSection = ({
```

File: src/components/Section/Section.test.js [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+  return renderWithIntl(component, translationsProperties);
+};
+
+describe('Section', () => {
+  it('should be rendered only with label', () => {
+    const { getByText } = renderSectionContainer(sectionWithLabel);
+
+    expect(getByText('test label')).toBeDefined();
+  });
+  it('should be rendered without headline', () => {
```
Comment (mariia-aloshyna) [inline src/components/Section/Section.test.js:61]: Separate test cases with an empty line, please
```suggestion
  });
  
  it('should be rendered without headline', () => {
```

File: src/components/Section/Section.test.js [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+    const { container } = renderSectionContainer(sectionWithoutHeadline);
+    const element = container.querySelector('h3');
```
Comment (mariia-aloshyna) [inline src/components/Section/Section.test.js:63]:
```suggestion
    const headlineElement = container.querySelector('h3');
```

File: src/components/Section/Section.test.js [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+
+    expect(element).toBeNull();
+  });
+  it('should be rendered without optional headline', () => {
+    const { container } = renderSectionContainer(sectionWithChildElements);
+    const element = container.querySelector('h3');
```
Comment (mariia-aloshyna) [inline src/components/Section/Section.test.js:69]:
```suggestion
    const headlineElement = container.querySelector('h3');
```

File: src/components/Section/Section.test.js [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+
+    expect(element).toBeNull();
+  });
+  it('should be rendered with data-test- attribute', () => {
+    const { container } = renderSectionContainer(sectionWithLabelAndDataTestAttribute);
+    const element = container.querySelector('[data-test-tester="true]');
```
Comment (mariia-aloshyna) [inline src/components/Section/Section.test.js:75]:
```suggestion
    const dataTestAttribute = container.querySelector('[data-test-tester="true]');
```

File: src/components/Section/Section.test.js [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+
+    expect(element).toBeDefined();
+  });
+  it('should render only with child elements', () => {
```
Comment (mariia-aloshyna) [inline src/components/Section/Section.test.js:79]:
```suggestion
  it('should be rendered only with child elements', () => {
```

File: src/components/Section/Section.test.js [hunk 10]
```diff
@@ -0,0 +1,105 @@
...
+    const { getByText } = renderSectionContainer(sectionWithChildElements);
+
+    expect(getByText('child component')).toBeDefined();
+  });
+  it('should render with label and disabled checkbox', () => {
```
Comment (mariia-aloshyna) [inline src/components/Section/Section.test.js:84]:
```suggestion
  it('should be rendered with label and disabled checkbox', () => {
```

File: src/components/Section/Section.test.js [hunk 11]
```diff
@@ -0,0 +1,105 @@
...
+  it('show children by clicking checkbox', () => {
+    const {
+      getByText,
+      getByRole,
+    } = renderSectionContainer(sectionWithCheckboxAndLabel);
+    const checkbox = getByRole('checkbox');
+
+    fireEvent.click(checkbox);
+
+    expect(getByText('child component')).toBeDefined();
```
Comment (mariia-aloshyna) [inline src/components/Section/Section.test.js:103]: Add one more check before `fireEvent.click(checkbox);` to ensure that children is not visible

**Review by mariia-aloshyna — APPROVED** [review]

**Review by IvanKryzhanovskyi — APPROVED** [review]

**Pull request merged**